am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.reform.amlib;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import uk.gov.hmcts.reform.amlib.exceptions.AccessManagementException;
+import uk.gov.hmcts.reform.amlib.repositories.AccessManagementRepository;
+
+public class AccessManagementService {
+    private Jdbi jdbi;
+
+    public AccessManagementService(Jdbi jdbi) {
+        this.jdbi = jdbi;
+
+        this.jdbi.installPlugin(new SqlObjectPlugin());
+    }
+
+    public void createResourceAccess(String resourceId, String accessorId) throws AccessManagementException {
+        Integer result;
+        try {
+            result = jdbi.withExtension(AccessManagementRepository.class, dao -> dao.createAccessManagementRecord(resourceId, accessorId));
+        } catch (Exception e) {
```

Comment:
what exception to you expect here? from the documentation http://jdbi.org/apidocs/org/jdbi/v3/core/Jdbi.html#withExtension-java.lang.Class-org.jdbi.v3.core.extension.ExtensionCallback- it may just throw http://jdbi.org/apidocs/org/jdbi/v3/core/extension/NoSuchExtensionException.html but that is so serious issue that we do not need to handle this in a special way, do we? so we could just ignore this catch here.

Reply:
I can also throw X:

```
throws NoSuchExtensionException,
                                                 X extends Exception
```